numpy/core/arrayprint.py

```diff
@@ -59,6 +59,10 @@ def set_printoptions(precision=None, threshold=None, edgeitems=None,
     ----------
     precision : int, optional
         Number of digits of precision for floating point output (default 8).
+        We raise a TypeError if this isn't of type int, as it is the first
+        arg provided, so is prone to being called incorrectly.
+        Without this set_printoptions wouldn't fail but subsequent calls
+        to print() would then fail with an opaque error.
```

Comment:
I don't think the explanation is necessary. When the type for an argument is declared as it is already, the caller ought to be prepared for a `TypeError` in any case. Or you can document the `TypeError` in a separate `Raises` section, but it doesn't need to be any more complicate than "A `TypeError` will be raised if any of the arguments are not of the correct type."